Lib/http/client.py

```diff
@@ -905,17 +905,23 @@ def _get_hostport(self, host, port):
                 host = host[:i]
             else:
                 port = self.default_port
-            if host and host[0] == '[' and host[-1] == ']':
-                host = host[1:-1]
+        if host and host[0] == '[' and host[-1] == ']':
+            host = host[1:-1]
 
         return (host, port)
 
     def set_debuglevel(self, level):
         self.debuglevel = level
 
+    def _wrap_ipv6(self, ip):
+        if ip.find(b':') >= 0 and ip.find(b'[') != 0:
```

Comment:
It can be optimized:
```suggestion
        if b':' in ip and ip[0] != b'['[0]:
```